servers/rendering/rasterizer_rd/shaders/particles.glsl

```diff
@@ -66,6 +66,38 @@ layout(set = 1, binding = 1, std430) restrict buffer Particles {
 }
 particles;
 
+#define EMISSION_FLAG_HAS_POSITION 1
+#define EMISSION_FLAG_HAS_ROTATION_SCALE 2
+#define EMISSION_FLAG_HAS_VELOCITY 4
+#define EMISSION_FLAG_HAS_COLOR 8
+#define EMISSION_FLAG_HAS_CUSTOM 16
+
+struct ParticleEmission {
+	mat4 xform;
+	vec3 velocity;
+	uint flags;
+	vec4 color;
+	vec4 custom;
+};
+
+layout(set = 1, binding = 2, std430) restrict volatile coherent buffer SourceEmission {
```

Comment:
Removing `volatile` and `coherent` qualifiers from here fixes shader compilation error, not sure what consequences this has. `GPUParticles3D` seems to be broken regardless of this change.

```suggestion
layout(set = 1, binding = 2, std430) restrict buffer SourceEmission {
```